### DISCUSSION TITLE
Week 5 - freeCodeCamp, Flexbox Froggy, and Project Thoughts

### DISCUSSION THREAD

**Discussion by coopsand** [USU-ITLS-x265-23-summer/class-repo#107] [General]
### What I LIKED:
- I liked that freeCodeCamp taught both CSS and HTML in one lesson. It allows you to see exactly how the two different files interact with each other.
- I liked that it provided feedback when you got the answer wrong. You were able to practice and improve your code.
- I felt that it didn't move too fast or two slow.
- Even with skipping a lesson in between this survey and the cafe menu, I found it was still easy to understand the goals of each step. I didn't have to look up anything that was covered in the colors lesson.
### What I DIDN'T LIKE:
- In a few spots, I felt like the instructions were written poorly. It took me a while to figure out where it wanted me to add ids and labels.
- The auto formatting sometimes formatted my code in a way that I didn't like.

### What I'd CHANGE:
The biggest thing I would change from freeCodeCamp would be just adding a prompt in that allows the user to specify if they wanted their code to be changed to fit the predetermined formatting or not. Sometimes freeCodeCamp put a lot on one line and it was tricky for me to see exactly where one item ended and the next began. If it would allow me to use more lines, I'd be more organized.

## Thoughts on Flexbox Froggy
I felt that Flexbox Froggy was pretty easy. I would like to see the addition of more levels like the final one. I still managed to complete that level in a reasonable amount of time. Maybe adding a more difficult gamemode would be a possibility? I felt that the game explained different terms well, and I was able to understand how to use them.

## Project Update
- **Portfolio**: With this project I feel that I would be able to implement a ton of different CSS features, and make the site feel custom to me. I'd like to see if I can learn how to add different pages to a site, just to make it more organized. I would use `html` to create different levels of headings, classes, and images. I'd also like to maybe implement a "contact me" page, where a user can fill out a survey with their email and inquiry - as a way to easily contact me. I don't know how I'd receive the results from this survey if I did decide to implement this into my site. (Recommendations?)
- **HEDI Website**: Similar to a portfolio, I want to learn how to create a multi-paged website. I feel this would be the best choice for keeping the site organized. I know how to change fonts, alignments, etc but would like to learn how to add in videos onto a website, since there is an existing video that I know would fit. I want to learn how I can best structure the homepage for this scenario. (I have a good idea regarding my portfolio, but this one I'm not so sure about.)  I'll use `html` to help structure different levels of the page, add images, and more.

**Comment by metasean** [comment]
> ... implement a "contact me" page ...

The best option for this class would simply be the [`mailto`](https://www.freecodecamp.org/news/mailto-link-how-to-make-an-html-email-link-example-code/) approach.  This will open the user's email app.   To have it not use the user's email app, requires a backend.  Setting up a backend is very much out of scope for this class.  There are some online services that you can use, but you will have to pay for them (e.g https://formspree.io/plans, https://mailthis.to/documentation.html#features).

**Reply by coopsand** [reply]
This is good to know!! Thanks!

**Comment by kassyoveson** [comment]
I agree that the coding camp was very unorganized. I had a hard time finding the right code to change at times. I started hitting enter to make more lines so I could read it better.

**Reply by coopsand** [reply]
That's what I ended up doing as well. It makes it so much easier to read and understand!

**Comment by gabebott** [comment]
I agree that the froggy game a bit simple, but I too thought the final level was a good challenge, and I wish there could be more levels at that difficulty
